### PR TITLE
[Linux] Fix incorrect texture format with OpenGL ES

### PR DESCRIPTION
**Pull request by huanghongxun**
This PR fixes [#76881](https://github.com/flutter/flutter/issues/76881).

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.
- [ ] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

**Comment by huanghongxun** [comment]
This pull request fixes embedder complains about render target creation failure after #24845 is applied.
Please note that this patch only fixes incorrect texture format. After applying this patch, the Flutter window unexpectedly whitens, which I have no idea for fixing.
@robert-ancell @wmww

File: shell/platform/linux/fl_backing_store_provider.cc [hunk 1]
```diff
@@ -78,7 +78,11 @@ uint32_t fl_backing_store_provider_get_gl_target(FlBackingStoreProvider* self) {
 }
 
 uint32_t fl_backing_store_provider_get_gl_format(FlBackingStoreProvider* self) {
-  return GL_RGBA8;
+  if (epoxy_is_desktop_gl()) {
```
Comment (robert-ancell) [inline shell/platform/linux/fl_backing_store_provider.cc:81]: Why are the formats different? This needs a comment to explain why each format is chosen.

File: shell/platform/linux/fl_renderer_gl.cc [hunk 2]
```diff
@@ -61,6 +61,9 @@ static gboolean fl_renderer_gl_create_backing_store(
   backing_store_out->open_gl.framebuffer.user_data = provider;
   backing_store_out->open_gl.framebuffer.name = name;
   backing_store_out->open_gl.framebuffer.target = format;
+  backing_store_out->open_gl.framebuffer.destruction_callback = [](void* p) {
```
Comment (robert-ancell) [inline shell/platform/linux/fl_renderer_gl.cc:64]: This change seems unrelated to the description of the commit - what is the purpose of it?

Comment (huanghongxun) [reply shell/platform/linux/fl_renderer_gl.cc:64]: Leaving destruction_callback unset will result in crash mentioned in the issue.

Comment (robert-ancell) [reply shell/platform/linux/fl_renderer_gl.cc:64]: The commit message doesn't mention this - it says "Fix incorrect texture format with OpenGL ES". Should this be two PRs?

Comment (huanghongxun) [reply shell/platform/linux/fl_renderer_gl.cc:64]: This line is no longer need, since https://github.com/flutter/engine/pull/24845 is merged.

Comment (robert-ancell) [reply shell/platform/linux/fl_renderer_gl.cc:64]: Can you remove it from the PR then please, thanks.

Comment (huanghongxun) [reply shell/platform/linux/fl_renderer_gl.cc]: Done.

File: shell/platform/linux/fl_backing_store_provider.cc (hunk 1 above)
Comment (huanghongxun) [reply shell/platform/linux/fl_backing_store_provider.cc:91]: Done.

Comment (huanghongxun) [reply shell/platform/linux/fl_backing_store_provider.cc:91]: But the reason is not explained in Skia. Comments here will only reference Skia code.

Comment (robert-ancell) [reply shell/platform/linux/fl_backing_store_provider.cc:91]: That's fine, referring to Skia at least leaves a trail to find the root cause.

**Review by robert-ancell — APPROVED** [review]
LGTM, thanks @huanghongxun!
